src/graph/GoExecutor.cpp

```diff
@@ -642,17 +653,10 @@ void GoExecutor::onVertexProps(RpcResponse &&rpcResp) {
     UNUSED(rpcResp);
 }
 
-std::vector<std::string> GoExecutor::getEdgeNamesFromResp(RpcResponse &rpcResp) const {
+std::vector<std::string> GoExecutor::getEdgeNames() const {
     std::vector<std::string> names;
     auto spaceId = ectx()->rctx()->session()->space();
-    auto &resp = rpcResp.responses();
-    auto *edgeSchema = resp[0].get_edge_schema();
-    if (edgeSchema == nullptr) {
-        return names;
-    }
-
-    for (auto &schema : *edgeSchema) {
-        auto edgeType = schema.first;
+    for (auto edgeType : edgeTypes_) {
```

Comment:
As i known, this function is used for supporting `over *`. According to the current implementation, i don't think it is appropriate to write it this way. In most cases you would not get all edge-type records. So this modification would make the output a wide-table which is sparse.

Reply:
Actually,  the logic is the same.  Further more,  if you extract edgeNames from edgeSchema in response, it is unordered.

Reply:
Got it.